File: android/src/main/java/name/zeno/android/presenter/bigbang/segment/SegmentUtil.java
/*
 * The MIT License (MIT)
 * Copyright (c) 2016 baoyongzhang <baoyz94@gmail.com>
 */
package name.zeno.android.presenter.bigbang.segment;

public class SegmentUtil
{
  public static String filterInvalidChar(String text)
  {
    return text.replaceAll("\\s*|\t|\r|\n", "");
  }

  public static boolean skipChar(char c)
  {
    return c == '\n' || c == '\r' || c == '\t' || c == ' ';
  }
}

File: library/src/main/java/name/zeno/android/presenter/bigbang/segment/SegmentUtil.kt
/*
 * The MIT License (MIT)
 * Copyright (c) 2016 baoyongzhang <baoyz94@gmail.com>
 */
package name.zeno.android.presenter.bigbang.segment

object SegmentUtil {
  fun filterInvalidChar(text: String): String = text.replace("\\s*|\t|\r|\n".toRegex(), "")
  fun skipChar(c: Char): Boolean = "\n\r\t ".contains(c)
}
